src/Impact.php

```diff
@@ -1301,12 +1301,37 @@ private static function addNode(array &$nodes, CommonDBTM $item): bool
         $image_name = $CFG_GLPI["impact_asset_types"][get_class($item)] ?? "";
         $image_name = self::checkIcon($image_name);
 
-        // Define basic data of the new node
+        $type = "";
+        if (class_exists($item->getType() . "Type")) {
+            $tabletype = getTableForItemType($item->getType() . "Type");
+            $typefield = getForeignKeyFieldForTable($tabletype);
+            $types_id = $item->fields[$typefield];
+            $type = Dropdown::getDropdownName($tabletype, $types_id);
+        }
+        $states_id = "";
+        if (isset($item->fields['states_id'])) {
+            $states_id = Dropdown::getDropdownName("glpi_states", $item->fields['states_id']);
+        }
+        $infocom = new Infocom();
+        $businesscriticities_id = "";
+        if ($infocom->getFromDBforDevice($item->getType(), $item->getID())) {
+            $businesscriticities_id
+                = Dropdown::getDropdownName(
+                    'glpi_businesscriticities',
+                    $infocom->fields['businesscriticities_id']
+                );
+        }
+       // Define basic data of the new node
```

Comment:
I'm afraid of the performance impact here for big graphes as it will be executed for each nodes.

What do you think @cedric-anne ?
Maybe the tooltip could be empty by default and then rendered using AJAX only when it is hovered like we have done in others places ?

Reply:
I do not know how this feature is used. If there is 20 assets displayed and each of them have these properties defined, indeed, it will add many requests.

Reply:
No problem for me with 25 objects in a graph for example